[pkg_npm] Tarball runfile does not respect workspace environment replacements

<!--🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅

Oh hi there! 😄 

To expedite issue processing please search open and closed issues before submitting a new one.
Existing issues often contain information about workarounds, resolution, or progress updates.

🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅-->


# 🐞 bug report

### Affected Rule

<!-- Can you pin-point one or more rules as the source of the bug? -->
<!-- ✍️edit: --> The issue is caused by the rule: `pkg_npm`

### Is this a regression?

Not to my knowledge, we had previously not attempted to use the `tgz` flag


### Description

Our team has a use case where we want to grab the generated tarball from the `pkg_npm` command so that we can augment it for distribution. After adding the `tgz` per rules_nodejs docs, we observe failures around our substitutions for `{VERSION}` (and/or workspace status command vars) not being replaced.

## 🔬 Minimal Reproduction

A minimal reproduction has been provided in [this sample repo](https://github.com/sumwatshade/bazel-simple) and found by running the following after cloning the repo:

```
yarn
yarn bazelisk build //packages/core:release.tar --verbose_failures --stamp
```

## 🔥 Exception or Error

<pre><code>
bazel-simple/packages/core/BUILD.bazel:83:8: Executing genrule //packages/core:release.tar failed: (Exit 1): bash failed: error executing command
  (cd /private/var/tmp/_bazel_lshadler/4105e2cfbecd5b0747c78df78ebf4cd8/execroot/bazel_simple && \
  exec env - \
    PATH=***** \
  /bin/bash -c 'source external/bazel_tools/tools/genrule/genrule-setup.sh; bazel-out/host/bin/packages/core/release.pack.sh 2>/dev/null | tail -1 | xargs -I {} cp {} bazel-out/darwin-fastbuild/bin/packages/core/core.tgz')
Execution platform: @local_config_platform//:host
Target //packages/core:release.tar failed to build
</code></pre>

Digging into this error and re-running the `release.pack.sh` command within the sandbox displays this:

<pre><code>
/private/var/tmp/_bazel_lshadler/4105e2cfbecd5b0747c78df78ebf4cd8/execroot/bazel_simple
❯ /bin/bash -c 'source external/bazel_tools/tools/genrule/genrule-setup.sh; bazel-out/host/bin/packages/core/release.pack.sh'
npm ERR! Invalid version: "${MY_VERSION}"

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/lshadler/.npm/_logs/2021-12-14T01_05_17_728Z-debug.log
</code></pre>


## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
macOS Big Sur 11.6.1
  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>
Bazelisk version: v1.10.1
Starting local Bazel server and connecting to it...
Build label: 4.2.2
Build target: bazel-out/darwin-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Thu Dec 2 18:28:52 2021 (1638469732)
Build timestamp: 1638469732
Build timestamp as int: 1638469732
  </code>
</pre>

**Rules_nodejs version:**

(Please check that you have matching versions between WORKSPACE file and `@bazel/*` npm packages.)

<pre>
  <code>
4.4.2
  </code>
</pre>

--- a/internal/pkg_npm/pkg_npm.bzl
+++ b/internal/pkg_npm/pkg_npm.bzl
@@ -395,13 +395,11 @@ def pkg_npm_macro(name, tgz = None, **kwargs):
         native.genrule(
             name = "%s.tar" % name,
             outs = [tgz],
-            # NOTE(mattem): on windows, it seems to output a buch of other stuff on stdout when piping, so pipe to tail
+            # NOTE(mattem): on windows, it seems to output a bunch of other stuff on stdout when piping, so pipe to tail
             # and grab the last line
             cmd = "$(location :%s.pack) 2>/dev/null | tail -1 | xargs -I {} cp {} $@" % name,
             srcs = [
                 name,
-            ],
-            tools = [
                 ":%s.pack" % name,
             ],
             tags = [

--- a/internal/pkg_npm/test/tgz_out/BUILD.bazel
+++ b/internal/pkg_npm/test/tgz_out/BUILD.bazel
@@ -7,6 +7,10 @@ pkg_npm(
         "main.js",
         "package.json",
     ],
+    stamp = "@rules_nodejs//nodejs/stamp:always",
+    substitutions = {
+        "0.0.0-PLACEHOLDER": "${BUILD_SCM_VERSION}",
+    },
     tgz = "my_tar.tgz",
 )
 

--- a/internal/pkg_npm/test/tgz_out/test.sh
+++ b/internal/pkg_npm/test/tgz_out/test.sh
@@ -17,6 +17,7 @@ function test_tgz_package_json() {
     tar -vxf "${TGZ}"
     
     assert_contains "awesome-package" "./package/package.json"
+    assert_contains "1.2.3" "./package/package.json"
     assert_contains "MAIN" "./package/main.js"
 }
 

